### ISSUE TITLE
Require spesific name for placeholder variables

### ISSUE THREAD

**Issue by valtlai** [eslint/eslint#13250]
**What rule do you want to change?**
[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)

**Does this change cause the rule to produce more or fewer warnings?**
More.

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new option, I think.

**Please provide some example code that this change will affect:**

```js
// This should be disallowed:
function (a, b) {
    // b is used, a is not
}

// And this allowed:
function (_, b) {
    // b is used, _ is not
}
```

**What does the rule currently do for this code?**
The unused variable `a` is allowed because it’s defined before the used function parameter.

**What will the rule do after it's changed?**
The first variable would be only allowed if it’s named in a spesific way (here `_`).

**Are you willing to submit a pull request to implement this change?**
Unfortunately I can’t.

**Comment by valtlai** [comment]
Just noticed there’s a option for that:

```json
"no-unused-vars": ["error", { "args": "all", "argsIgnorePattern": "^_$" }]
```

So closing.
